app/controllers/recipes_controller.rb

```diff
@@ -15,23 +15,26 @@ def web_result
 
   # GET /recipes or /recipes.json
   def index
-    @recipes = Recipe.with_text(params[:query]).in_categories(params[:category_names]).sorted
+    @recipes = policy_scope(Recipe).with_text(params[:query]).in_categories(params[:category_names]).sorted
   end
 
   # GET /recipes/1 or /recipes/1.json
-  def show; end
+  def show
+    authorize @recipe
+    @recipe = Recipe.find(params[:id])
```

Comment:
This isn't needed, `@recipe` was already set by the `set_recipe` before action.  It also wouldn't make sense to set the recipe after calling authorize, because you could end up setting a different recipe.

I think you could handle all of the `authorize` calls in this controller by updating the `set_recipe` method:

```
def set_recipe
  @recipe = authorize Recipe.find(params[:id])
end
```